src/averageAges.js

```diff
@@ -9,12 +9,30 @@
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
+ *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
+  const menArr = people.filter((person) => person.sex === 'm');
+  const arrAge = menArr.reduce((accumulator, man) => {
+    if (century && Math.ceil(man.died / 100) !== century) {
+      return accumulator;
+    }
+
+    const age = man.died - man.born;
+
+    accumulator.push(age);
+
+    return accumulator;
+  }, []);
+
+  const averageAge = arrAge
+    .reduce((accumulator, age) => accumulator + age) / arrAge.length;
```

Comment:
you can create separate function for this calculation, to not to repeat this code in each function

Reply:
comment wasn't fixed